drivers/gpio/gpio_tle9104.c

```diff
@@ -0,0 +1,544 @@
+/*
+ * Copyright (c) 2023 SILA Embedded Solutions GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT infineon_tle9104
+
+#include <errno.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/init.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/gpio/gpio_utils.h>
+#include <zephyr/drivers/spi.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/byteorder.h>
+
+LOG_MODULE_REGISTER(gpio_tle9104, CONFIG_GPIO_LOG_LEVEL);
+
+/*
+ * The values for the defines below as well as the register definitions were
+ * taken from the datasheet, which can be found at:
+ * https://www.infineon.com/dgdl/Infineon-TLE9104SH-DataSheet-v01_31-EN.pdf?fileId=5546d462766cbe86017676144d76581b
+ */
+#define TLE9104_RESET_DURATION_TIME_IN_US                    10
+#define TLE9104_RESET_DURATION_WAIT_TIME_SAFETY_MARGIN_IN_US 200
+#define TLE9104_RESET_DURATION_WAIT_TIME_IN_US               10
+#define TLE9104_GPIO_COUNT                                   4
+#define TLE9104_INITIALIZATION_TIMEOUT_IN_MS                 1
```

Comment:
nit: time defines suffixes are normally just the constant, like `TIMEOUT_MS` not `TIMEOUT_IN_MS`, there's few cases with the `IN` as well but I suggest sticking to the shorter version.... it's... shorter :-)

Reply:
done